public/app/engagementindex/leaderboard/leaderboardController.js

```diff
@@ -0,0 +1,299 @@
+                             'Upper Quartile: {point.q3}<br/>' +
+                             'Median: {point.median}<br/>' +
+                             'Lower Quartile: {point.q1}<br/>' +
+                             'Minimum: {point.low}',
+                'borderColor': 'transparent'
+              }
+            // Current user points
+            }, {
+              'data': [[0, $scope.me.points]],
+              'marker': {
+                'fillColor': '#3179BC',
+                'lineWidth': 5,
+                'lineColor': '#3179BC'
+              },
+              'tooltip': {
+                'headerFormat': '',
+                'pointFormat': 'My Points: {point.y}'
+              },
+              'type': 'scatter'
+            }
+          ]
+        });
+      }, 0);
+    };
+
+    /**
+     * Calculate the maximum value, upper quartile, median, lower quartile
+     * and minimum value of a data series. These values can be used to
+     * generate a box plot diagram
+     *
+     * @param  {Number[]}     series          The data series used to calculate the box plot values
+     * @return {Number[]}     boxplotData     The calculated box plot values
+     * @return {Number}       boxplotData[0]  The maximum value
+     * @return {Number}       boxplotData[1]  The upper quartile value
+     * @return {Number}       boxplotData[2]  The median value
+     * @return {Number}       boxplotData[3]  The lower quartile value
+     * @return {Number}       boxplotData[4]  The minimum value
+     */
+    var calculateBoxPlotData = function(series) {
+      // Sort the provided data series in ascending order
+      series.sort(function(a, b) {
+        return b - a;
+      });
+
+      var min = series[0];
+      var max = series[series.length - 1];
+
+      // Calculate the quartiles
+      var q1 = calculateQuartile(series, 1);
+      var q2 = calculateQuartile(series, 2);
+      var q3 = calculateQuartile(series, 3);
+
+      return [max, q3, q2, q1, min];
+    };
+
+    /**
+     * Calculate a quartile value for an ordered data series
+     *
+     * @param  {Number[]}     series        The data series to calculate a quartile value for. Note that this should be sorted in ascending order
+     * @param  {Number}       quartile      The quartile to generate. For the median value, `2` should be provided
+     * @return {Number}                     The calculated quartile
+     */
+    var calculateQuartile = function(series, quartile) {
+      // Calculate the position of the quartile point in the series
+      var quartileIndex = Math.floor(series.length / 4 * quartile);
+
+      // If the quartile point lands on an item in the series, return that value
+      if (series.length % 2) {
+        return series[quartileIndex];
+      // If the quartile point lands in between 2 items, calculate the average of those items
+      } else {
+        return (series[quartileIndex - 1] + series[quartileIndex]) / 2;
```

Comment:
Does a quartile need to be an observable event? I didn't think that had to be the case